modules/swagger-codegen/src/main/resources/python/api_client.mustache

```diff
@@ -625,3 +640,124 @@ class ApiClient(object):
             if klass_name:
                 instance = self.__deserialize(data, klass_name)
         return instance
+ 
+{{#httpsig}}    
+    def get_sha256_digest(self, data):
+        """
+        :param data: Data set by User
+        :return: instance of digest object 
+        """
+        digest = SHA256.new()
+        digest.update(data)
+        
+        return digest
+    
+    def get_rsasig_b64encode(self, private_key_path, digest):
+        """
+        :param private_key_path : abs path to private key .pem file.
+        :param digest: digest
+        :return: instance of digest object 
+        """
+        
+        key = open(private_key_path, "r").read()
```

Comment:
Opening a key file on every request is a hit for performance.

Also there is no error handling at all. What if file reading fails?

Reply:
Constantly reading the file also bothered me too. I was thinking about how to handle this and the problem that auth_settings is coded to be a static, hard-coded set of values driven by the swagger spec; that's not typically how people setup security. So I'm going to do a bunch of changes to the code now that I've dug into a bit, so it will read the file once and cache it in a mutable auth-settings. That will also provide people the ability to store the cert data instead of having to have it at rest on disk. 


 (Wasn't my code, just got drafted to contribute back to the ecosystem, because I wanted to contribute something else). Feel free to give as much feedback as you want.